File: email-service/src/main/java/com/github/manosbatsis/services/email/model/UserEvent.java
package com.github.manosbatsis.services.email.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class UserEvent {

    private UserEventKey key;

    private String type;
    private String data;
}

File: event-service/src/main/kotlin/com/github/manosbatsis/services/event/model/UserEvent.kt
package com.github.manosbatsis.services.event.model

import org.springframework.data.cassandra.core.mapping.PrimaryKey
import org.springframework.data.cassandra.core.mapping.Table

@Table("user_events")
data class UserEvent(
    @PrimaryKey val key: UserEventKey,
    val type: String,
    val data: String,
)
